src/Board.hs

```diff
@@ -60,14 +62,14 @@ scrambleBoard board@(Board (Dimensions width height) _) startingBoard gen =
                in verifyBoard newBoard gen iterations
           verifyBoard board gen iterations
               | board == startingBoard = scrambleBoardLoop board gen iterations
-              | otherwise              = scrambleBoardLoop board gen (pred iterations)
+              | otherwise              = scrambleBoardLoop board gen (iterations - 1)
```

Comment:
I see your point on the explicit subtraction for iterations.  I actually debated that internally for a bit, so you get the sway vote. :)